CitadelExtras/WeaponBob.cs

```diff
@@ -0,0 +1,64 @@
+﻿using UnityEngine;

```

Comment:
Would this be attached to the ReloadDipContainer to ensure affect applies to all weapons? If so, then there'd be a conflict in control with the WeaponFire script (on the GunCamera) causing oddness during reload and recoil.  Should link to the WeaponFire script or add this to that script's Update.

Although, I suppose you could have a copy of this on each individual weapon and set differing sway amounts on each though, but would need to store its current localPosition as the default in Start and apply the offset to that position as they are all positioned differently from 0 0 0.

Reply:
Hmm, thinking about it more, the separate script attached to each weapon (or its subcontainer as referenced in WeaponCurrent), would add more data set in the editor, making this type of thing more diffult to mod after the fact.  Ideally it would be set and saved as a part of WeaponFire (or maybe it was WeaponCurrent, whichever does reload) such that it could be changed in build.  I've been tending more and more towards data living outside of the editor whereever possible to better facilitate the use of a "trainer" program for stats-based modding, should someone want to make such a thing someday.  In this case, an array of 16 sway strength values, indexed using the "wep16" index, and making the sway occur as part of the recoil and reload lerping would allow for swaying the ReloadDipContainer GameObject and probably keep this pretty simple, and (bonus!) use the same Update method to minimize the overhead of this feature.

Reply:
okay I probably should have explained that better. "Works for most weapons" just means that if you copy the script and tweak some things like offsets it will still function. Right now it doesn't work on the pipe. 

The script is attached to the w_pistol/v_whatgunyoureholding objects. The one I uploaded won't work with anything, it's just a rough template. I'll upload a better version that is meant for the pistol. I also need to edit the formula so it's an infinity symbol instead of a circle.